core/domain/question_jobs_one_off_test.py

```diff
@@ -79,7 +80,7 @@ def test_migration_job_does_not_convert_up_to_date_question(self):
         self.assertEqual(question.question_state_data.to_dict(),
                          updated_question.question_state_data.to_dict())
 
-        output = question_jobs_one_off.QuestionMigrationOneOffJob.get_output(job_id) # pylint: disable=line-too-long
+        output = question_jobs.QuestionMigrationOneOffJob.get_output(job_id)
```

Comment:
```
        output = (
            question_jobs_one_off.QuestionMigrationOneOffJob.get_output(job_id))
```

or, if necessary:

```
        output = (
            question_jobs_one_off.QuestionMigrationOneOffJob.get_output(
                job_id))
```

Ditto elsewhere below, where needed.

Reply:
I agree that this is a possible solution. 

But I have recently started seeing some repeated patterns in our codebase that seem to reduce the readability/scannability, more specifically that we are being too explicit/redundant at the expense of readability.

Because the developer (probably) knows or can easily check that he is in a file named question_jobs_**one_off**_test.py, then he sees a statement `        
```
output = (
    question_jobs_one_off.QuestionMigrationOneOffJob.get_output(
        job_id))
```
which repeats both `one_off`/`OneOff` and `Job`/`jobs` twice. That means that the fact that he is testing a one-off job is repeated almost thrice. For me personally, having the whole statement on one line is much more readable than split into two or three lines.

I think that the best solution **here** would be probably to rename `QuestionMigrationOneOffJob` to `QuestionMigration`, also we might consider the usage of `as` statements in imports for at least the test files.

Reply:
Thanks for the analysis! Here are some thoughts:

- I think the redundancy is a valid concern. However, I wonder if it's a general enough pattern. For example, in this case, the words "question", "one-off" and "job" are repeated. I think it's reasonable to repeat "question" here, though -- since you could conceivably imagine jobs in this file that migrate sub-parts of questions and they would be named SubpartMigrationOneOffJob instead. So, that leaves us with "one-off" and "job", and those are generic and potentially removable.
- However, if we make changes/renamings, they should be general in nature. So, if we drop the "OneOffJob" suffix, we should do that for all one-off jobs, not just this one.
- I think class names should stand on their own, so, I would keep "Job" (the same way we keep "Model" in ExplorationModel). But, I think it's fine to remove "OneOff" for the reasons you state; just make sure there isn't any place where these might get confused with "continuous" jobs. We should do these for all jobs though (to maintain consistency), so maybe do it in a separate PR.
- I am against expanding the use of "as", since that means we end up having multiple names for the same thing across the codebase, which is confusing. Also, I am not keen on a restriction along the lines of "do XXX only in test files" for an XXX that isn't obviously related to tests.

Please let me know if you have any thoughts on the above or would like to discuss this further. Thanks!

Reply:
I mostly agree, with all the points.

> just make sure there isn't any place where these might get confused with "continuous" jobs

In most (all?) places we import the module and then use the classes, thus this wouldn't happen because a one-off job would be used as `question_jobs_one_off.QuestionMigrationJob` and continuous job as `question_jobs_continuous.QuestionMigrationJob`. 

Which brings me to another alternative: `from question_jobs_one_off import QuestionMigrationOneOffJob` to import the class, which would also make sense, but I'm not sure how keen are we on using `from` imports.

Reply:
OK, thanks! Yep, that makes sense.

To the last alternative, I think I'm not very keen -- we prefer to import at the module level.